pkg/udf/forward/forward.go

```diff
@@ -753,3 +750,24 @@ func errorArrayToMap(errs []error) map[string]int64 {
 	}
 	return result
 }
+
+// check if the options provided are for a valid map mode
+// exactly one of the appliers should not be nil as only one mode can be active at a time, not more not less only 1
+// the way our map mode options work is that whenever one of the applier is set, it sets the others appliers to nil.
+// There is a case where none of them is set which cannot be allowed
+func isValidMapMode(opts *options) bool {
+	// if all the appliers are set, then it is an invalid scenario
+	if (opts.batchMapUdfApplier != nil) && (opts.unaryMapUdfApplier != nil) && (opts.streamMapUdfApplier != nil) {
+		return false
+	}
```

Comment:
I think this case is covered by the if statement below, can we remove it?

Reply:
The below check is a like a 3 input XOR gate 
`a XOR b XOR c `
When all of them are true -> the output will still be true, hence we need this extra check here

Example is here
https://go.dev/play/p/ha6pnfsq79u

Reply:
Ah, it took me quite a while to figure this out and I was still wrong...

nit: in this case, maybe using a single counter to check only one is non-nil is more readable. Feel free to disagree.